back/app/Http/Controllers/ItemController.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Item;
+use App\Http\Requests\ItemIndexRequest;
+use App\Http\Requests\ItemStoreRequest;
+use Illuminate\Support\Facades\Storage;
+use Illuminate\Support\Facades\DB;
+use Exception;
+
+class ItemController extends Controller
+{
+    public function index(ItemIndexRequest $request)
+    {
+        try{
+            $validated = $request->validated();
+            $query = Item::query();
+            //検索
+            if (isset($validated['name']) && $validated['name'] !== '') {
+                $query->where('name', 'like', '%' . $validated['name'] . '%');
+            }
+            //絞り込み
+            if (isset($validated['effectType'])) {
+                $query->where('effect_type', $validated['effectType']);
+            }
+            //並び替え
+            $sortOptions = [
+                'createdAt' => 'created_at',
+                'updatedAt' => 'updated_at',
+                'name' => 'name',
+            ];
+            $sortColumn = $sortOptions[$validated['sort']];
+            $sortDirection = $validated['desc'] ? 'desc' : 'asc';
+            $query->orderBy($sortColumn, $sortDirection);
+            //ページネーション
+            $currentPage = $validated['currentPage'] ?? 1;
+            $items = $query->paginate(40, ['*'], 'page', $currentPage);
+
+            $formattedItems = $items->map(function ($item) {
+                return[
+                    'id' => $item->id,
+                    'name' => $item->name,
+                    'imageUrl' => $item->image_url,
+                    'effectType' => $item->effect_type,
+                ];
+            });
+            return response()->json([
+                'items' => $formattedItems
+            ]);
```

Comment:
- jsonで直接返す
- haaderで合計ページを返す

```suggestion
            $formattedItems = $items->map(function ($item) {
                return[
                    'id' => $item->id,
                    'name' => $item->name,
                    'imageUrl' => $item->image_url,
                    'effectType' => $item->effect_type,
                ];
            });
            return response()->json($formattedItems)
              ->header('X-TOTAL-PAGE', $totalPage);
;
```